src/makeInfinityAdder.js

```diff
@@ -3,8 +3,26 @@
 /***
  * @return {function}
  */
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(num) {
+  let sum = num || 0;
+  let sumReserve = 0;
+
+  function result(nextNum) {
+    sumReserve = 0;
```

Comment:
The `sumReserve` variable is unnecessary. The `sum` variable itself can be returned and then reset to 0 when the function is called without arguments.